factory/ideas.php

```diff
@@ -265,9 +253,29 @@ public function get_idea($id)
 	}
 
 	/**
+	 * Returns an idea specified by its topic ID.
+	 *
+	 * @param int $id The ID of the idea to return.
+	 *
+	 * @return array The idea
+	 */
+	public function get_idea_by_topic_id($id)
```

Comment:
Right, it's just one tiny extra query... We eliminated 3 other queries though, so still a net savings going on ;)

An alternative may be to add topic_id param to the get_idea function, and search be either idea_id or topic_id but that can be a different PR if we really want it.